File: articles/hope-theory-positive-psychology/go/hope_indices.go
package main

import "fmt"

type HopeIndicators struct {
	AgencyScore         float64
	PathwaysScore       float64
	GoalClarity         float64
	GoalProgress        float64
	WellbeingScore      float64
	MeaningScore        float64
	StressLoad          float64
	ObstacleIntensity   float64
	SocialSupport       float64
	ResourceAccess      float64
	GoalRevisionQuality float64
	ContextSupport      float64
}

func HopeIndex(x HopeIndicators) float64 {
	return (x.AgencyScore + x.PathwaysScore) / 2.0
}

func ContextSupportIndex(x HopeIndicators) float64 {
	return (x.SocialSupport + x.ResourceAccess + x.ContextSupport) / 3.0
}

func NetPathwayContext(x HopeIndicators) float64 {
	return x.PathwaysScore + ContextSupportIndex(x) + x.GoalRevisionQuality - x.ObstacleIntensity
}

func NetFutureOrientation(x HopeIndicators) float64 {
	return x.AgencyScore +
		x.PathwaysScore +
		x.GoalClarity +
		x.GoalProgress +
		x.MeaningScore +
		ContextSupportIndex(x) -
		x.StressLoad -
		x.ObstacleIntensity
}

func main() {
	example := HopeIndicators{7.1, 7.0, 7.3, 6.8, 7.1, 7.4, 3.5, 3.7, 7.2, 6.8, 6.6, 7.0}
	fmt.Printf("Hope index: %.3f\n", HopeIndex(example))
	fmt.Printf("Context support index: %.3f\n", ContextSupportIndex(example))
	fmt.Printf("Net pathway context: %.3f\n", NetPathwayContext(example))
	fmt.Printf("Net future orientation: %.3f\n", NetFutureOrientation(example))
}

File: articles/hope-theory-positive-psychology/fortran/hope_indices.f90
program hope_indices
  implicit none
  real :: agency_score, pathways_score, goal_clarity, goal_progress
  real :: wellbeing_score, meaning_score, stress_load, obstacle_intensity
  real :: social_support, resource_access, goal_revision_quality, context_support
  real :: hope, context_support_index, net_pathway_context, net_future_orientation

  agency_score = 7.1
  pathways_score = 7.0
  goal_clarity = 7.3
  goal_progress = 6.8
  wellbeing_score = 7.1
  meaning_score = 7.4
  stress_load = 3.5
  obstacle_intensity = 3.7
  social_support = 7.2
  resource_access = 6.8
  goal_revision_quality = 6.6
  context_support = 7.0

  hope = (agency_score + pathways_score) / 2.0
  context_support_index = (social_support + resource_access + context_support) / 3.0
  net_pathway_context = pathways_score + context_support_index + goal_revision_quality - obstacle_intensity
  net_future_orientation = agency_score + pathways_score + goal_clarity + goal_progress + &
      meaning_score + context_support_index - stress_load - obstacle_intensity

  print *, "Hope index:", hope
  print *, "Context support index:", context_support_index
  print *, "Net pathway context:", net_pathway_context
  print *, "Net future orientation:", net_future_orientation
end program hope_indices
